scripts/bumpfee.py

```diff
@@ -0,0 +1,254 @@
+parser.add_option('--yes',
+                  action='store_true',
+                  dest='answeryes',
+                  default=False,
+                  help='answer yes to everything')
+add_base_options(parser)
+
+def check_valid_candidate(orig_tx, orig_tx_info, wallet):
+    # check that the transaction is still unconfirmed
+    if orig_tx_info['confirmations'] > 0:
+        raise RuntimeWarning('Transaction already confirmed. Nothing to do.')
+
+    # all transaction inputs must belong to the wallet
+    own_inputs_n = len(wallet.inputs_consumed_by_tx(orig_tx))
+    tx_inputs_n = len(orig_tx.vin)
+
+    if own_inputs_n != tx_inputs_n:
+        raise ValueError('Transaction inputs should belong to the wallet.')
+
+    # at least one input should signal opt-in rbf
+    if not any([vin.nSequence <= 0xffffffff - 2 for vin in orig_tx.vin]):
+        raise ValueError('Transaction not replaceable.')
+
+    # wallet should own at least one output that we can change values for
+    if not any(
+        [wallet.is_known_script(vout.scriptPubKey) for vout in orig_tx.vout]
+    ):
+        raise ValueError('Transaction has no output we can deduct fees from.')
+
+    # wallet inputs must be in the same mixdepth - sanity check
+    _mixdepths = set()
+    for ix in range(len(orig_tx.vin)):
+        _mixdepths.add(
+            wallet.get_details(
+                wallet.script_to_path(
+                    wallet.pubkey_to_script(
+                        btc.extract_pubkey_from_witness(orig_tx, ix)[0]
+                    )
+                )
+            )[0]
+        )
+    if len(_mixdepths) != 1:
+        raise ValueError('Transaction inputs must be from the same mixdepth.')
+
+def tx_vsize(tx):
+    raw_tx_size = len(tx.serialize())
+    witness_size = len(tx.wit.serialize())
+    non_witness_size = raw_tx_size - witness_size
+    return int(non_witness_size + .25 * witness_size)
+
+def compute_bump_fee(tx, tx_info, fee_rate):
+    tx_size_n = tx_vsize(tx)
+    tx_fee = btc.amount_to_sat(abs(tx_info['fee']))
+    proposed_fee_rate = fee_rate / Decimal(1000.0)
+    proposed_fee = int(tx_size_n * proposed_fee_rate)
+    min_proposed_fee = tx_fee + tx_size_n
+    min_proposed_fee_rate = min_proposed_fee / Decimal(tx_size_n)
+
+    if proposed_fee < (tx_fee + tx_size_n):
+        raise ValueError('Proposed fee for transaction replacement: '
+                         '%d sats (%.1f sat/vB) is below minimum required '
+                         'for relay: %d sats (%.1f sat/vB). '
+                         'Try using a higher fee setting.'
+                         % (proposed_fee, proposed_fee_rate, 
+                            min_proposed_fee, min_proposed_fee_rate))
+
+    return (proposed_fee - tx_fee)
+
+def prepare_transaction(new_tx, old_tx, wallet):
+    input_scripts = {}
+    spent_outs = []
+    for ix, vin in enumerate(new_tx.vin):
+        script = wallet.pubkey_to_script(
+            btc.extract_pubkey_from_witness(old_tx, ix)[0])
```

Comment:
`extract_pubkey_from_witness` only supports `p2wpkh`. We need to support timelocked (but now spendable) utxos and also ideally the old `p2sh-p2wpkh`.